xapian-applications/omega/templates/query

```diff
@@ -106,7 +106,7 @@ $if{$field{size},<span title="$html{$field{size}} bytes">Size: <b>$html{$filesiz
 </div>
 </div>
 </td>
-<td><b><a href="$html{$field{url}}">$html{$or{$field{caption},$field{title},$prettyurl{$field{url}},Untitled}}</a></b><br>
+<td><b><a href="$html{omega.cgi?FMT=$url{clicklog}&QUERYID=$url{$hash{md5,$transform{s/^\s+|\s+$$//g,\1,$query}}}&DOCID=$url{$id}&URL=$url{$field{url}}}">$html{$or{$field{caption},$field{title},$prettyurl{$field{url}},Untitled}}</a></b><br>
```

Comment:
The terms listed in `xP` are compared to those from parsing the current query and there are four cases:

```
// Heuristics:
// * If any terms have been removed, it's a "fresh query" so we discard any
//   relevance judgements
// * If all previous terms are there but more have been added then we keep
//   the relevance judgements, but return the first page of hits
//
// NEW_QUERY entirely new query
// SAME_QUERY unchanged query
// EXTENDED_QUERY new query, but based on the old one
// BAD_QUERY parse error (message in error_msg)
typedef enum { NEW_QUERY, SAME_QUERY, EXTENDED_QUERY, BAD_QUERY } querytype;
```

The criteria for clearing relevance judgements are weaker than those for forcing the first page - we assume if terms are added to an existing query that it's still the same "information need" so documents already marked as relevant should be kept.